gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -239,7 +240,13 @@ private WriteObjectResponse doResumableUpload() throws IOException {
 
       // Wait for streaming RPC to become ready for upload.
       try {
-        responseObserver.ready.await();
+        // wait for 1 min for the channel to be ready. Else bail out
+        if (!responseObserver.ready.await(60*1000, MILLISECONDS)) {
```

Comment:
let's add a test to cover the path, lgtm otherwise

Reply:
I tried looking into the test code and am not able to figure out how to trigger this failure. can you help @mprashanthsagar

Reply:
IMO, the issue seems that we are not doing countDown for ready on error. What happens if error happens before start itself?

Reply:
@abmodi Do you mean error from the api/connector ? 

If it is from the API, that should trigger the onError callback. Failing that the timeout should kick-in. 

If it is connector, that exception should bubble up if we have not reached the latch, If blocked on latch, timeout should kick-in again

Reply:
The responseObserver error is called after the channel is established. In this case we are observing that the channel establishment itself just hangs. There is no path in the code where onError is called while establishing the channel (IIRC)

Reply:
Sounds good.